File: ai25btech11031/Assignments/Matgeo/3.3.15/Codes/solution.c
#include <stdio.h>
#include <math.h>
#include "solution.h"

int main() {
    double BC, AD, angleA;
    printf("Enter length BC: ");
    scanf("%lf", &BC);
    printf("Enter length of median AD: ");
    scanf("%lf", &AD);
    printf("Enter angle A (in degrees): ");
    scanf("%lf", &angleA);

    // Fix B at (0,0) and C at (BC,0)
    Point B = {0, 0};
    Point C = {BC, 0};
    Point D = {(B.x + C.x)/2, (B.y + C.y)/2};

    // Circle with center D and radius AD
    double r1 = AD;

    // Angle condition: circumcircle with radius BC/(2*sinA)
    double A_rad = angleA * M_PI / 180.0;
    double R = BC / (2.0 * sin(A_rad));

    // Two possible centers of the angle-locus circle
    Point O1 = {D.x, D.y + R};
    Point O2 = {D.x, D.y - R};

    Point p1, p2;

    printf("\nPossible coordinates of A:\n");

    // Check intersections for O1
    if (circle_intersections(D, r1, O1, R, &p1, &p2)) {
        printf("A1 = (%.3f, %.3f)\n", p1.x, p1.y);
        printf("A2 = (%.3f, %.3f)\n", p2.x, p2.y);
    }

    // Check intersections for O2
    if (circle_intersections(D, r1, O2, R, &p1, &p2)) {
        printf("A3 = (%.3f, %.3f)\n", p1.x, p1.y);
        printf("A4 = (%.3f, %.3f)\n", p2.x, p2.y);
    }

    return 0;
}

File: ai25btech11031/Assignments/Matgeo/3.3.15/Codes/solution.py
import math

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

def circle_intersections(c1, r1, c2, r2):
    dx = c2.x - c1.x
    dy = c2.y - c1.y
    d = math.hypot(dx, dy)

    if d > r1 + r2 or d < abs(r1 - r2) or d == 0:
        return []  # no intersection

    a = (r1*r1 - r2*r2 + d*d) / (2*d)
    h = math.sqrt(max(r1*r1 - a*a, 0))
    xm = c1.x + a*dx/d
    ym = c1.y + a*dy/d
    rx = -dy * (h/d)
    ry = dx * (h/d)

    p1 = Point(xm + rx, ym + ry)
    p2 = Point(xm - rx, ym - ry)

    return [p1, p2]

def main():
    # Input
    BC = float(input("Enter length BC: "))
    AD = float(input("Enter length of median AD: "))
    angleA = float(input("Enter angle A (in degrees): "))

    # Points B, C, D
    B = Point(0, 0)
    C = Point(BC, 0)
    D = Point((B.x + C.x) / 2, (B.y + C.y) / 2)

    # Circle 1: center D, radius AD
    r1 = AD

    # Angle condition: circumcircle with radius BC/(2*sinA)
    A_rad = math.radians(angleA)
    R = BC / (2.0 * math.sin(A_rad))

    # Two possible centers of angle-locus circle
    O1 = Point(D.x, D.y + R)
    O2 = Point(D.x, D.y - R)

    print("\nPossible coordinates of A:")

    # Intersections with O1 circle
    sol1 = circle_intersections(D, r1, O1, R)
    for i, p in enumerate(sol1, start=1):
        print(f"A{i} = ({p.x:.3f}, {p.y:.3f})")

    # Intersections with O2 circle
    sol2 = circle_intersections(D, r1, O2, R)
    for i, p in enumerate(sol2, start=len(sol1)+1):
        print(f"A{i} = ({p.x:.3f}, {p.y:.3f})")

if __name__ == "__main__":
    main()

